src/components/shared/forms/editUserForm.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react'
+
+import { User, ExpressValidationError } from 'devu-shared-modules'
+
+import { useAppDispatch } from 'redux/hooks'
+import { SET_ALERT } from 'redux/types/active.types'
+
+import TextField from 'components/shared/inputs/textField'
+
+import RequestService from 'services/request.service'
+
+type Props = {
+  user: User
+}
+
+const EditUserForm = ({ user }: Props) => {
+  const [formData, setFormData] = useState<User>(user)
+  const setAlert = useAppDispatch(SET_ALERT)
+
+  const handleUpdatePreferredName = (preferredName: string) => setFormData({ ...formData, preferredName })
+
+  const handleSubmit = () => {
+    RequestService.put(`/api/users/${user.id}`, formData)
+      .then(() => {
+        setAlert({ autoDelete: false, type: 'success', message: 'User Preferences Updated' })
+      })
+      .catch((err: ExpressValidationError[] | Error) => {
+        const message = Array.isArray(err) ? err.map((e) => `${e.param} ${e.msg}`).join(', ') : err.message
+        setAlert({ autoDelete: false, type: 'error', message })
+      })
+  }
+
+  return (
+    <div>
+      <h1>User Information</h1>
+      <TextField
+        onChange={handleUpdatePreferredName}
+        label='Preferred Name'
+        id='preferredName'
+        placeholder={user.preferredName}
+      />
+      <TextField type='email' label='Email' id='email' placeholder={user.email} disabled />
+      <TextField label='Person Number' id='externalId' placeholder={user.externalId} disabled />
+      <button onClick={handleSubmit}>Submit</button>
```

Comment:
We should probably make this a button component. That way we're not using raw buttons anywhere

Reply:
Oh one more thing we might want to add is a semi transparent loading overlay for when this request is saving. That way they can only click save once and there's clearly something happening for them on save